### ISSUE TITLE
Unhandled rejection Error: gm convert: Unable to open file (travis-ci)

### ISSUE THREAD

**Issue by maxcnunes** [electron-userland/electron-builder#320]
electron-builder: `^3.5.2`

I have [this](https://github.com/sqlectron/sqlectron-gui/blob/upgrade-builder/.travis.yml) travis configuration (which was based on onshape-desktop-shell).

For some reason, the same configuration that I can run locally on my Ubuntu. On travis does not work ([full buid log](https://travis-ci.org/sqlectron/sqlectron-gui/jobs/123493855)).

```
CI detected, so artifacts will be published if draft release exists
Installing app dependencies for arch x64 to /home/travis/build/sqlectron/sqlectron-gui/app
----------------------------------------------------
Reading icns family from /home/travis/build/sqlectron/sqlectron-gui/build/icon.icns...
  Saved 'is32' element to /tmp/electron-builder-4421G5iIHRFgWJn8/icon_16x16x32.png.
  Saved 'il32' element to /tmp/electron-builder-4421G5iIHRFgWJn8/icon_32x32x32.png.
  Saved 'ih32' element to /tmp/electron-builder-4421G5iIHRFgWJn8/icon_48x48x32.png.
  Saved 'it32' element to /tmp/electron-builder-4421G5iIHRFgWJn8/icon_128x128x32.png.
 Extracted 4 elements from /home/travis/build/sqlectron/sqlectron-gui/build/icon.icns.
Unhandled rejection Error: gm convert: Unable to open file (/tmp/electron-builder-4421G5iIHRFgWJn8/icon_256x256x32.png) [No such file or directory].
Error: Command failed: gm convert -size 96x96 /tmp/electron-builder-4421G5iIHRFgWJn8/icon_256x256x32.png -resize 96x96 /tmp/electron-builder-4421G5iIHRFgWJn8/icon_96x96x32.png
gm convert: Unable to open file (/tmp/electron-builder-4421G5iIHRFgWJn8/icon_256x256x32.png) [No such file or directory].
    at /home/travis/build/sqlectron/sqlectron-gui/node_modules/electron-builder/src/util.ts:85:18
    at ChildProcess.exithandler (child_process.js:209:5)
    at emitTwo (events.js:100:13)
    at ChildProcess.emit (events.js:185:7)
    at maybeClose (internal/child_process.js:850:16)
    at Socket.<anonymous> (internal/child_process.js:323:11)
    at emitOne (events.js:90:13)
    at Socket.emit (events.js:182:7)
    at Pipe._onclose (net.js:475:12)
From previous event:
    at Object.exec (/home/travis/build/sqlectron/sqlectron-gui/node_modules/electron-builder/src/util.ts:72:10)
    at resize (/home/travis/build/sqlectron/sqlectron-gui/node_modules/electron-builder/src/linuxPackager.ts:106:18)
    at LinuxPackager.<anonymous> (/home/travis/build/sqlectron/sqlectron-gui/node_modules/electron-builder/src/linuxPackager.ts:109:62)
    at [object Generator].next (native)
From previous event:
    at tsAwaiter (/home/travis/build/sqlectron/sqlectron-gui/node_modules/electron-builder/src/awaiter.ts:10:51)
    at Object.build (/home/travis/build/sqlectron/sqlectron-gui/node_modules/electron-builder/src/builder.ts:30:59)
    at Object.<anonymous> (/home/travis/build/sqlectron/sqlectron-gui/node_modules/electron-builder/src/build-cli.ts:41:2)
    at Module._compile (module.js:413:34)
    at Object.Module._extensions..js (module.js:422:10)
    at Module.load (module.js:357:32)
    at Function.Module._load (module.js:314:12)
    at Function.Module.runMain (module.js:447:10)
    at startup (node.js:146:18)
    at node.js:404:3
```

Does anyone has any idea why this is happening?

**Comment by develar** [comment]
`icns2png` (`libicns` package) is broken on Travis. As you are a Linux guru ;),  you can try to find what package source (to get updated version of this tool) or package should be installed on Travis to fix it.

As of for me, I am going to bundle icns2png for Linux since it is a broken.

Workaround: " (or you can put them into the build/icons directory — filename must contains size (e.g. 32x32.png))." See https://github.com/develar/onshape-desktop-shell/tree/master/build/icons for example.

**Comment by develar** [comment]
Maybe installing package `libjasper1` will help.

**Comment by maxcnunes** [comment]
After a lot of attempts. I got this to work using the Ubuntu Trusty version - https://docs.travis-ci.com/user/trusty-ci-environment/.

```yaml
sudo: required
dist: trusty
```

Thanks @develar .

**Comment by develar** [comment]
`sudo` is not an option because it disables caching on Travis. Installing `libopenjpeg5` doesn't help?

Or cache works?

Thanks a lot for testing and information.

**Comment by develar** [comment]
Ok, I see in your repo such attempts to use libopenjpeg5.

Reopened — must be fixed, current behaviour is not user friendly.

**Comment by maxcnunes** [comment]
Yeah, I tried installing `libopenjpeg5` in the default Linux distro on Travis. But seems `libopenjpeg5` is not available in that distro.
I have also tried removing the `sudo` argument. But seems the distro is not changed without that argument.
Even if using the sudo argument it disables the caching on Travis. The last successful build took 3min 15secs. So I don't mind losing the cache since the build is working and the build still in a pretty good time.
